scripts/commands/checkPatternsIndex.ts

```diff
@@ -0,0 +1,212 @@
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { readFile } from "fs/promises";
+import { pathExists } from "../lib/fs";
+
+const IGNORE_URL = ["/guides/qiskit-serverless"];
+
+const INDEX_PAGES = [
+  "docs/guides/map-problem-to-circuits.mdx",
+  "docs/guides/optimize-for-hardware.mdx",
+  "docs/guides/execute-on-hardware.mdx",
+  "docs/guides/postprocess-results.mdx",
+];
+
+const TOC_PATH = `docs/guides/_toc.json`;
+
+async function getIndexEntries(indexPage: string): Promise<string[]> {
+  const rawIndex = await readFile(indexPage, "utf-8");
+  const result: string[] = [];
+  for (const line of rawIndex.split("\n")) {
+    // The index represents an unordered list of entries starting with `*`.
+    if (!line.trimStart().startsWith("* ")) {
+      continue;
+    }
+
+    const slug = extractPageSlug(line);
+    if (slug) {
+      result.push(slug);
+    }
+  }
+
+  return result;
+}
+
+function extractPageSlug(text: string): string | undefined {
+  const re = /\((.*)\)/gm;
+  // Ex: '* [Circuit library](./circuit-library)'.
+  const match = re.exec(text);
+  if (!match) {
+    // Nested sections don't have any link
+    return undefined;
+  }
+  const pageSlug = match[1];
+  if (pageSlug.startsWith("http") || pageSlug.startsWith("/")) {
+    return pageSlug;
+  }
+  return `/guides/${pageSlug.split("/").pop()}`;
+}
+
+function getTocSectionPageNames(sectionNode: any): string[] {
+  let results = [];
+  if (sectionNode.url) {
+    results.push(sectionNode.url);
+  }
+
+  if (sectionNode.children) {
+    for (const child of sectionNode.children) {
+      results = [...results, ...getTocSectionPageNames(child)];
+    }
+  }
+
+  return results;
+}
+
+async function getToolsTocEntriesToCheck(): Promise<string[]> {
+  const toc = JSON.parse(await readFile(TOC_PATH, "utf-8"));
+  const toolsNode = toc.children.find((child: any) => child.title == "Tools");
+  const toolsPages = getTocSectionPageNames(toolsNode);
+  return toolsPages.filter((page) => !IGNORE_URL.includes(page));
+}
+
+async function getDeduplicateEntriesAndAddErrors(
+  src: string,
+  errors: string[],
+): Promise<string[]> {
+  const entries =
+    src == TOC_PATH
+      ? await getToolsTocEntriesToCheck()
+      : await getIndexEntries(src);
+  const deduplicatedPages: string[] = [];
+
+  for (const entry of entries) {
+    if (deduplicatedPages.includes(entry)) {
+      errors.push(`❌ ${src}: The entry ${entry} is duplicated`);
+    } else {
+      deduplicatedPages.push(entry);
+    }
+  }
+
+  return deduplicatedPages;
+}
+
+function addExtraIndexPagesErrors(
+  indexPage: string,
+  indexEntries: string[],
+  toolsEntries: string[],
+  errors: string[],
+): void {
+  const ExtraIndexPages = indexEntries.filter(
+    (page) => !toolsEntries.includes(page),
+  );
+  if (ExtraIndexPages.length > 0) {
```

Comment:
Why the if statement? You can also do `return indexEntries.filter().map()` directly